definitions/procedures/foreman/remove_obsolete_roles.rb

```diff
@@ -0,0 +1,29 @@
+module Procedures::Foreman
+  class RemoveObsoleteRoles < ForemanMaintain::Procedure
+    metadata do
+      for_feature :foreman_database
+      description 'Remove obsolete roles that are not being used by any user or usergroup'
+      confine do
+        check_min_version('foreman', '1.11')
+      end
+    end
+
+    def run
+      duplicate_role_ids = feature(:foreman_database).query(
+        Checks::Foreman::CheckDuplicateRoles.query_to_get_duplicate_roles
```

Comment:
Is it possible to avoid calling SELECT query twice ? maybe we can store the results in check and pass it to procedure if there are any duplicate roles ?

Reply:
As using foreman-maintain, it is possible to run procedure directly that's why I have not added parameterized procedure.